Food-Waste-Optimization/Food-Waste-Optimization
# Authentication

Nearly all files from /frontend changed:

-Switched from react-router 6.23 -> 6.7 to make routes function with Microsoft authentication
-Had to modify all component files to create different views for authenticated/non-authenticated users
-Modified main.jsx and App.jsx to integrate MS authentication and new version of react-router
-Added 3 files to /utils - folder: 
    - authConfig.js contains configuration for authentication
    - NavigationClient.js is needed for custom navigation to combine react-router and ms-authentication
    - claimUtils.js was used in ms-tutorial for debugging, kept it there just in case it would be still needed. Can be removed later.

To test these changes:
  - Create .env to React root (/frontend)
  - Copy the contents to it from slack #technical
  - run npm install before starting (make sure react-router-dom get upgraded to 6.7)
  - Credentials for log in can be found from slack #ilmoitukset
